e2e/rapid_test.go

```diff
@@ -0,0 +1,598 @@
+package e2e
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/0xPolygon/pbft-consensus"
+	"go.opentelemetry.io/otel/trace"
+	"golang.org/x/sync/errgroup"
+	"io"
+	"log"
+	"pgregory.net/rapid"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+)
+
+const waitDuration = 50 * time.Millisecond
+
+type fakeTransport struct {
```

Comment:
Could we maybe reuse some of the existing implementation in e2e framework? We have similar implementation with `genericGossipTransport` (with it you are able to attach custom callback which can be used as a filter for messages delivery).

Reply:
I thought about it. I didn't manage to do it simply. So I created a separate issue for it https://linear.app/matic/issue/POS3-184/property-tests-toolkit-refactoring